mmeds/database/data_uploader.py

```diff
@@ -0,0 +1,181 @@
+        self.mdata = MMEDSDoc(created=datetime.utcnow(),
+                              last_accessed=datetime.utcnow(),
+                              testing=self.testing,
+                              doc_type='sequencing_run',
+                              reads_type=self.reads_type,
+                              barcodes_type=self.barcodes_type,
+                              study_name=self.sequencing_run_name,
+                              access_code=self.access_code,
+                              owner=self.owner,
+                              public=self.public)
+
+        self.mdata.save()
+
+        count = 0
+        new_dir = fig.SEQUENCING_DIR / ('{}_{}_{}'.format(self.owner, self.sequencing_run_name, count))
+
+        while new_dir.is_dir():
+            count += 1
+            new_dir = fig.SEQUENCING_DIR / ('{}_{}_{}'.format(self.owner, self.sequencing_run_name, count))
+        new_dir.mkdir()
+
+        self.path = Path(new_dir) / 'database_files'
+        MMEDSDoc.objects.timeout(False)
+
+    def get_info(self):
+        """ Method to return a dictionary of relevant info for the process log """
+        info = {
+            'created': self.created,
+            'type': 'upload-run',
+            'owner': self.owner,
+            'pid': self.pid,
+            'name': self.name,
+            'exitcode': self.exitcode
+        }
+        return info
+
+    def run(self):
+        """
+        Thread that handles the upload of sequencing run files.
+        """
+        self.mdata.update(is_alive=True)
+        self.mdata.save()
+        Logger.debug('Handling upload for sequencing run {} for user {}'.format(self.sequencing_run_name, self.owner))
+
+        # If the owner is None set user_id to 0
+        if self.owner is None:
+            self.user_id = 0
+            self.email = fig.MMEDS_EMAIL
+        # Otherwise get the user id for the owner from the database
+        else:
+            sql = 'SELECT user_id, email FROM user WHERE user.username=%(uname)s'
+            cursor = self.db.cursor()
+            cursor.execute(sql, {'uname': self.owner})
+            result = cursor.fetchone()
+            cursor.close()
+            # Ensure the user exists
+            if result is None:
+                raise NoResultError('No account exists with the provided username and email.')
+            self.user_id = int(result[0])
+            self.email = result[1]
+
+        # If the metadata is to be made public overwrite the user_id
+        if self.public:
+            self.user_id = 1
+        self.check_file = fig.DATABASE_DIR / 'last_check.dat'
+
+        if not self.path.is_dir():
+            self.path.mkdir()
+        self.mdata.update(path=str(self.path.parent))
+        self.mdata.save()
+
+        # Create a copy of the Data files
+        datafile_copies = {key: create_local_copy(Path(filepath).read_bytes(),
+                                                  f"{key}.fastq.gz", self.path.parent, False)
```

Comment:
You are making assumptions about the file format here (`.fastq.gz`), should not be harcoded.

Reply:
I'll parameterize this, although at present note that in fact this is the only type that we can handle. this PR does explicitly break the already-demultiplexed ZIP upload type, which I believe was the only other type we could use. I will make an issue about fixing that

Reply:
Not needed to fix now, and yes, mostlikely this will be the only type we use. But it's always good to make the least number of assumptions. Create an issue is sufficient to resolve this.